pkg/river/scanner/scanner_test.go

```diff
@@ -0,0 +1,249 @@
+package scanner
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/grafana/agent/pkg/river/token"
+	"github.com/stretchr/testify/assert"
+)
+
+type tokenExample struct {
+	tok token.Token
+	lit string
+}
+
+var tokens = []tokenExample{
+	// Special tokens
+	{token.COMMENT, "/* a comment */"},
+	{token.COMMENT, "# a comment \n"},
+	{token.COMMENT, "// a comment \n"},
+	{token.COMMENT, "/*\r*/"},
+	{token.COMMENT, "/**\r/*/"}, // golang/go#11151
+	{token.COMMENT, "/**\r\r/*/"},
+	{token.COMMENT, "#\r\n"},
+	{token.COMMENT, "//\r\n"},
+
+	// Identifiers and basic type literals
+	{token.IDENT, "foobar"},
+	{token.IDENT, "a۰۱۸"},
+	{token.IDENT, "foo६४"},
+	{token.IDENT, "bar９８７６"},
+	{token.IDENT, "ŝ"},    // golang/go#4000
+	{token.IDENT, "ŝfoo"}, // golang/go#4000
+	{token.NUMBER, "0"},
+	{token.NUMBER, "1"},
+	{token.NUMBER, "123456789012345678890"},
+	{token.NUMBER, "01234567"},
+	{token.FLOAT, "0."},
+	{token.FLOAT, ".0"},
+	{token.FLOAT, "3.14159265"},
+	{token.FLOAT, "1e0"},
+	{token.FLOAT, "1e+100"},
+	{token.FLOAT, "1e-100"},
```

Comment:
Nitpick so just disregard this for now but our grammar wouldn't allow this; exponents are only available on floats, which _must_ have a dot in them.